src/main/java/lotto/domain/Lotto.java

```diff
@@ -33,41 +53,30 @@ private static String[] getSplit(String value) {
         return value.split(",");
     }
 
-    public Lotto(Integer... numbers) {
-        this(new TreeSet<>(Arrays.asList(numbers)));
-    }
-
-    public Lotto(Set<Integer> numbers) {
-        validate(numbers);
-        this.numbers = Collections.unmodifiableSet(numbers);
-    }
-
-    private void validate(Set<Integer> numbers) {
+    private void validate(Set<LottoNumber> numbers) {
         if (numbers.size() != LOTTO_NUMBER_SIZE) {
             throw new IllegalArgumentException();
         }
-
-        boolean invalidNumber = numbers.stream()
-                .anyMatch(n -> n < 1 || n > 45);
-        if (invalidNumber) {
-            throw new IllegalArgumentException();
-        }
     }
 
     public int match(Lotto winningLotto) {
         int count = 0;
-        for (Integer number : numbers) {
+        for (LottoNumber number : numbers) {
             if (winningLotto.numbers().contains(number)) {
```

Comment:
```suggestion
            if (winningLotto.contains(number)) {
```
이미 contains() 메서드도 추가되어 있음